step4__generate_tsv_of_cram_paths.py

```diff
@@ -1,71 +1,81 @@
-len(set(current_samples_v31) - set(v31_release_samples))   # 919 samples  
-
-ht = hl.read_table("gs://gnomad-bw2/gnomad_v3_readviz_crams__that_failed_AB_filter_exploded_keyed_by_sample.ht")
-current_samples_v3 = ht.distinct().S.collect()  # 68,639 samples
-
-len(set(current_samples_v3) - set(v31_release_samples)) # 1,655
-
-"""
-
-#%%
-
-tsv_paths = hl.hadoop_ls("gs://gnomad-bw2/gnomad_v3_1_readviz_tsvs")
-
-#%%
-
-path_tuples = [(os.path.basename(t['path']).replace(".tsv.bgz", ""), t['path']) for t in tsv_paths]
-df = pd.DataFrame(path_tuples, columns=['entity:participant_id', 'variants_tsv_bgz'])
-df = df.set_index('entity:participant_id')
-
-#%%
+import asyncio
+import argparse
+import logging
 
-df2 = pd.read_table("./metadata/v3_1_new_releasable_cram_paths_with_sex.txt").rename(columns={'CRAM': 'cram_path', 'CRAI': 'crai_path'})
-df2 = df2[['sample_id', 'cram_path', 'crai_path']]
-df2 = df2.set_index('sample_id')
-
-#%%
-
-df_final = df.join(df2, how="inner")
-assert sum(df_final.variants_tsv_bgz.isna()) == 0
-assert sum(df_final.cram_path.isna()) == 0
-
-df_final.columns
-
-#%%
-
-df_final.reset_index().rename(
-    columns={'index': 'sample_id'}).to_csv(
-    "step4_output__cram_and_tsv_paths_table.tsv", index=False, header=True, sep="\t")
-
-
-#%%
-
-#%%
-df_final = df_final.reset_index().rename(columns={'index': 'entity:participant_id'})
-df_final.to_csv("step4_output__terra_tsv_paricipants.tsv", index=False, header=True, sep="\t")
-
-#%%
-
-df_final['membership:participant_set_id'] = "sample_set_" + datetime.datetime.now().strftime("%Y_%m_%d")
-df_final['participant'] = df_final["entity:participant_id"]
-df_final = df_final[['membership:participant_set_id', "participant"]]
-df_final.to_csv("step4_output__terra_tsv_participant_set.tsv", index=False, header=True, sep="\t")
+import hail as hl
 
-#%%
+from gnomad.resources.resource_utils import DataException
+from gnomad.utils.file_utils import parallel_file_exists
+
+from .utils import get_sample_ids
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("get_tsv_for_haplotype_caller")
+logger.setLevel(logging.INFO)
+
+
+def main(args):
+    """
+    Generate single TSV with sample IDs, cram paths, and variant TSV file paths.
+
+    NOTE: Run locally (`parallel_file_exists` only works locally).
+    """
+    output_bucket = args.output_bucket
+
+    logger.info("Getting sample IDs...")
+    sample_ids = get_sample_ids(args.ids_file)
+
+    tsvs = [f"{output_bucket}/{sample}.tsv.bgz" for sample in sample_ids]
+    tsv_file_exists = asyncio.get_event_loop().run_until_complete(
+        parallel_file_exists(tsvs)
+    )
+
+    logger.info("Starting cram existence checks...")
+    cram_map = {}
+    with hl.hadoop_open(args.cram_map) as c:
+        for line in c:
+            sample, cram = line.strip().split("\t")
+            cram_map[sample] = cram
+    crams_exist = asyncio.get_event_loop().run_until_complete(
```

Comment:
would use a consistent name for parallel_file_exists return object (`crams_exist` vs. `tsv_file_exists`)